lib/charms/operator_libs_linux/v1/systemd.py

```diff
@@ -66,149 +71,146 @@
 class SystemdError(Exception):
     """Custom exception for SystemD related errors."""
 
-    pass
 
+def _systemctl(*args: str, check: bool = False) -> Optional[int]:
+    """Control a system service using systemctl.
 
-def _popen_kwargs():
-    return {
-        "stdout": subprocess.PIPE,
-        "stderr": subprocess.STDOUT,
-        "bufsize": 1,
-        "universal_newlines": True,
-        "encoding": "utf-8",
-    }
+    Args:
+        *args: Arguments to pass to systemctl.
```

Comment:
For this to render properly in the docs, is the proper annotation for this `args:` or `*args:`? I'm guessing the former, but I'm not sure.

Reply:
Hmm... from a quick search online, there does not seem to be clear consensus on the proper annotation for `*args` in docstrings. Anecdotally, the docstrings for the `dnf` charm library seem to render fine on Charmhub: https://charmhub.io/operator-libs-linux/libraries/dnf. Several of the methods in the dnf charm library use `*packages` in the docstring for documenting the `*packages` argument.

Reply:
Interesting. One thing that's weird is that they render as:

```python
def install()
```

Whereas I would have expected something like this:

```python
def install(*packages)
```

As is, it looks like the function doesn't take any arguments, until you scan down to the argument section, and then it's confusing. Maybe that's because of the `*packages` name? Or maybe that's some other issue. I'm not sure what tool we're using to generate that on Charmhub.io.